src/utils/time.ts

```diff
@@ -1,17 +1,28 @@
-export interface TimeComponents {
-    hours?:number;
-    minutes?:number;
-    seconds?:number;
-    ms?:number;
+enum ConcreteTimeComponents {
```

Comment:
Ach, nice! Really oldschool pattern here, I completely forgot about that over all those fancy modern language features. Shame on me, repeat offense will result in revocation of my "Old Man" license.